.github/workflows/dev-releases-for-pr.yml

```diff
@@ -0,0 +1,76 @@
+name: Release Dev Build for PR
+
+on:
+  pull_request:
+    types:
+      - opened
+      - synchronize
+
+jobs:
+  setup:
+    if: github.event.pull_request.draft == false
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        os: [macos-12]
+    timeout-minutes: 30
+    outputs:
+      plugins: ${{ steps.packages.outputs.paths }}
+    steps:
+      - name: Cancel Previous Runs
+        uses: styfle/cancel-workflow-action@ce177499ccf9fd2aded3b0426c97e5434c2e8a73
+        with:
+          access_token: ${{ secrets.GITHUB_TOKEN }}
+      - name: Get Latest
+        uses: actions/setup-node@v3
+        with:
+          node-version: 18.18.0
+      - uses: actions/checkout@v3
+      - name: Restore Dependency Cache
+        id: cache-modules
+        uses: actions/cache@v3
+        with:
+          path: |
+            node_modules
+            */node_modules
+          key: ${{ runner.os }}-dependency-caching-${{ hashFiles('package.json', '*/package.json') }}
+      - run: npm install
+      - id: files
+        uses: tj-actions/changed-files@v37
+        with:
+          json: true
+          write_output_files: true
+      - id: catjson
+        run: |
+          echo "FILES_JSON=$(cat .github/outputs/all_changed_files.json)" >> $GITHUB_OUTPUT
+      - id: packages
+        uses: ./.github/actions/changed-packages
+        with:
+         files: ${{ steps.catjson.outputs.FILES_JSON }}
+
+  release-dev-of-plugins:
+    runs-on: macos-12
+    timeout-minutes: 30
+    needs:
+      - setup
+    strategy:
+      matrix:
+        plugin: ${{ fromJson(needs.setup.outputs.plugins) }}
+    steps:
+      - uses: actions/setup-node@v3
+        with:
+          node-version: 18.18.0
+      - uses: actions/checkout@v3
+      - name: Restore Dependency Cache
+        id: cache-modules
+        uses: actions/cache@v3
+        with:
+          path: |
+            node_modules
+            */node_modules
+          key: ${{ runner.os }}-dependency-caching-${{ hashFiles('package.json', '*/package.json') }}
+      - run: npm install
+      - run: | 
+          npm version prerelease --no-git-tag-version -f --preid dev-$()-$(date +\"%Y%m%dT%H%M%S\")
```

Comment:
Should we maybe include part of the SHA to make it easier to determine what's what? Or maybe the PR number? Or both?